app/form_models/copy_vacancy_form.rb

```diff
@@ -12,7 +12,7 @@ def initialize(vacancy:)
     self.job_title = vacancy.job_title
     self.starts_on = vacancy.starts_on
     self.ends_on = vacancy.ends_on
-    self.expires_on = vacancy.expires_on
+    self.expires_on = vacancy.expires_on.future? ? vacancy.expires_on : nil
```

Comment:
I'm wondering if it would be simpler not to set any date fields if the vacancy is expired. The vacancy could have expired 1 day ago or more than a year ago so if we try to be helpful in pre populating the dates that _would_ pass validation I worry we'll have to keep a copy of the all the date validation logic here to do it right. 

For example the `starts_on` must be after the `ends_on` if present, and both must be after the `publish_on`.

Would it be simpler and safer to say if a listing has expired then we can't trust any of its dates so instead we ask the user fill out all the dates again, only 2 of which are mandatory.

Perhaps:

```
  def initialize(vacancy:)
    self.vacancy = vacancy
    self.job_title = vacancy.job_title
    prepopulate_date_fields
  end

  def prepopulate_date_fields
    return if vacancy.expires_on.past?

    self.starts_on = vacancy.starts_on
    self.ends_on = vacancy.ends_on
    self.expires_on = vacancy.expires_on
    self.publish_on = vacancy.publish_on
  end
```

Reply:
I [added a commit](https://github.com/dxw/teacher-vacancy-service/pull/731/commits/b719b8660b61e09a51dcf540ddabf004a54d9d42) that explicitly sets the dates to `nil` if the vacancy has expired.

I couldn't do your suggestion as the date fields get delegated the the vacancy instance.

Reply:
Of course 🤦‍♂️